frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleAssetPlugin.ts

```diff
@@ -0,0 +1,67 @@
+import * as webpack from 'webpack';
+import * as path from 'path';
+import { ConsolePackageJSON } from '../schema/plugin-package';
+import { ConsoleExtensionsJSON } from '../schema/console-extensions';
+import { ConsolePluginManifestJSON } from '../schema/plugin-manifest';
+import { ExtensionValidator } from '../validation/ExtensionValidator';
+import { SchemaValidator } from '../validation/SchemaValidator';
+import { extensionsFile, pluginManifestFile } from '../constants';
+import consoleExtensionsSchema from '../../dist/schema/console-extensions';
+
+const validateExtensionsFile = (ext: ConsoleExtensionsJSON) => {
+  return new SchemaValidator(extensionsFile).validate(consoleExtensionsSchema, ext);
+};
+
+const emitJSON = (compilation: webpack.Compilation, filename: string, data: any) => {
+  const source = JSON.stringify(data, null, 2);
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  compilation.emitAsset(filename, {
```

Comment:
You should make it a habit wherever you disable rules to add a comment as to why.

Reply:
OK, will do. In this particular case, the `emitAsset` webpack API requires `Source` argument with lots of mandatory methods, but most aren't really required or necessary at runtime, so it's essentially a webpack typing issue.